File: skipad-collector/src/com/skipad/collector/common/IResourceFile.java
package com.skipad.collector.common;

public interface IResourceFile {
	
	long getId();
    String getPath();
    String getContentType();
    Integer getWidth();
    Integer getHeight();
    Integer getBitrate();
    String getExtension();
    long getResourceId();
    ResourceFilePathType getPathType();
    boolean isThumbnail();
    
    public static class ColumnNames
    {
        public static final String ResourceFileId = "ResourceFileId";
        public static final String Path = "Path";
        public static final String ContentType = "ContentType";
        public static final String Width = "Width";
        public static final String Height = "Height";
        public static final String Bitrate = "Bitrate";
        public static final String Extension = "Extension";
        public static final String InsertDate = "InsertDate";
        public static final String ResourceId = "ResourceId";
		public static final String PathTypeId = "PathTypeId";
		public static final String IsThumbnail = "IsThumbnail";
    }
    
}

File: Skipad.Common/IResourceFile.cs
using Jayrock.Json;

namespace Inqwise.Skipad.Common
{
    public enum ResourceFilePathType
    {
        Undefined = 0,
        Relative = 1,
        Absolute = 2,
    }

    public interface IResourceFile
    {
        long Id { get; }
        string Path { get; }
        string ContentType { get; }
        int? Width { get; }
        int? Height { get; }
        int? Bitrate { get; }
        string Extension { get; }
        JsonObject ToJson();
        long ResourceId { get; }
        ResourceFilePathType PathType { get; }
        IResource Resource { get; }
        IResourceFile Thumbnail { get; }
    }
}